python/pyarrow/_s3fs.pyx

```diff
@@ -234,6 +234,8 @@ cdef class S3FileSystem(FileSystem):
             S3FileSystem(proxy_options={'scheme': 'http', 'host': 'localhost',
                                         'port': 8020, 'username': 'username',
                                         'password': 'password'})
+    allow_delayed_open : bool, default False
+        Whether to allow file-open methods to return before the actual open.
```

Comment:
Can we improve this a bit by explaining the consequences, as in the corresponding C++ docstring?